### PR TITLE
add mtmalign

### PR DESCRIPTION
**Pull request by lrauschning**
This PR adds a module for the structural aligner mTM-align, for use in the multiplesequencealignment pipeline.

## PR checklist
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/mtmalign/align/main.nf [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    output:
+    tuple val(meta), path("./mTM_result/result.fasta")  , emit: alignment
+    tuple val(meta), path("./mTM_result/result.pdb")    , emit: structure
+    path "versions.yml"                                 , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    """
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/main.nf:24]:
```suggestion
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
```

File: modules/nf-core/mtmalign/align/main.nf [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    ls *.pdb | sed s/\\ /\\n/ > input_list.txt
+    mtm-align -i input_list.txt
+
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/main.nf:27]:
```suggestion
mv ./mTM_result/result.fasta ${prefix}.aln
mv ./mTM_result/result.pdb ${prefix}.pdb
```

File: modules/nf-core/mtmalign/align/main.nf [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    conda "bioconda::mtm-align=20220104"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mtm-align:20220104--h4ac6f70_0':
+        'biocontainers/mtm-align:20220104--h4ac6f70_0' }"
+
+    input:
+    tuple val(meta), path('*.pdb', arity: '2..*')
+
+    output:
+    tuple val(meta), path("./mTM_result/result.fasta")  , emit: alignment
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/main.nf:16]:
```suggestion
    tuple val(meta), path("./mTM_result/*.aln")  , emit: alignment
```

File: modules/nf-core/mtmalign/align/main.nf [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    tuple val(meta), path("./mTM_result/result.pdb")    , emit: structure
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/main.nf:17]:
```suggestion
    tuple val(meta), path("./mTM_result/*.pdb")    , emit: structure
```

File: modules/nf-core/mtmalign/align/main.nf [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        mTM-align: \$( mtm-align -h | grep -e "\\(Version [[:digit:]]*\\)" | grep -oe "[[:digit:]]*" )
+    END_VERSIONS
+    """
+
+    stub:
+    """
+    mkdir mTM_result
+    touch mTM_result/result.fasta
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/main.nf:38]:
```suggestion
    touch mTM_result/${prefix}.aln
    touch ./mTM_result/${prefix}.pdb
```

File: modules/nf-core/mtmalign/align/main.nf [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    # mtm-align -v prints the wrong version 20180725, so extract it from the cosmetic output in the help message
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        mTM-align: \$( mtm-align -h | grep -e "\\(Version [[:digit:]]*\\)" | grep -oe "[[:digit:]]*" )
+    END_VERSIONS
+    """
+
+    stub:
+    """
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/main.nf:36]:
```suggestion
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
```

File: modules/nf-core/mtmalign/align/meta.yml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+output:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'test']`
+  - alignment:
+      type: file
+      description: Alignment in FASTA format.
+      pattern: "result.fasta"
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/meta.yml:38]:
```suggestion
      pattern: "*.aln"
```

File: modules/nf-core/mtmalign/align/meta.yml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  - structure:
+      type: file
+      description: Overlaid structures in PDB format.
+      pattern: "result.pdb"
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/meta.yml:42]:
```suggestion
      pattern: "*.pdb"
```

File: modules/nf-core/mtmalign/align/tests/main.nf.test [hunk 9]
```diff
@@ -0,0 +1,48 @@
+//include { UNTAR } from '../../../../../modules/nf-core/untar/main.nf'
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/tests/main.nf.test:1]:
```suggestion
```

File: modules/nf-core/mtmalign/align/tests/main.nf.test [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+            then {
+                assert process.success
+                // mTMalign may be nondeterministic, just check if the pdbs are all in there
+                //assert snapshot(process.out).match()
+                assert file(process.out.alignment[0][1]).text.contains(">1.pdb")
+                assert file(process.out.alignment[0][1]).text.contains(">2.pdb")
+                assert file(process.out.alignment[0][1]).text.contains(">3.pdb")
+                assert file(process.out.alignment[0][1]).text.contains(">4.pdb")
+                assert file(process.out.alignment[0][1]).text.contains(">5.pdb")
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/tests/main.nf.test:45]:
```suggestion
                assertAll(
                    { assert process.success  },
                    // mTMalign may be nondeterministic, just check if the pdbs are all in there
                    //assert snapshot(process.out).match()
                    { assert file(process.out.alignment[0][1]).text.contains(">1.pdb")  },
                    { assert file(process.out.alignment[0][1]).text.contains(">2.pdb")  },
                    { assert file(process.out.alignment[0][1]).text.contains(">3.pdb") },
                    { assert file(process.out.alignment[0][1]).text.contains(">4.pdb") },
                    { assert file(process.out.alignment[0][1]).text.contains(">5.pdb") },
                    { assert snapshot(process.out.versions).match("versions") }
```

**Review by luisas — CHANGES_REQUESTED** [review]
Super nice Leon!!

I just added a couple of comments - mainly it comes down to allowing the user to use the prefix to change the name of the output file :)

File: modules/nf-core/mtmalign/align/main.nf [hunk 11]
```diff
@@ -0,0 +1,46 @@
+
+
+process MTMALIGN_ALIGN {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "bioconda::mtm-align=20220104"
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/main.nf:7]: Should we also update it to this? 

https://github.com/nf-core/modules/blob/0f4f5205a67d3a90adc8e341cf2b19a16f5cb272/modules/nf-core/muscle5/super5/main.nf#L4

Comment (lrauschning) [reply modules/nf-core/mtmalign/align/main.nf:7]: true, we should – I'm not sure its documented anywhere, but it is apparently now the preferred way to have conda envs.

File: modules/nf-core/mtmalign/align/main.nf (hunk 4 above)
Comment (lrauschning) [reply modules/nf-core/mtmalign/align/main.nf:17]: (mtmalign copies the input structures into the results directory)

File: modules/nf-core/mtmalign/align/main.nf [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    ls *.pdb | sed s/\\ /\\n/ > input_list.txt
+    mtm-align -i input_list.txt
+mv ./mTM_result/result.fasta ${prefix}.aln
+mv ./mTM_result/result.pdb ${prefix}.pdb
```
Comment (JoseEspinosa) [inline modules/nf-core/mtmalign/align/main.nf:30]:
```suggestion
    mv ./mTM_result/result.fasta ${prefix}.aln
    mv ./mTM_result/result.pdb ${prefix}.pdb
```

**Review by JoseEspinosa — COMMENTED** [review]
LGTM! Please fix test and address @luisas's comment and it will be ready to merge.

**Comment by lrauschning** [comment]
Anybody know what's up with the linting there?
```
Conflicting process name between environment.yml (mtmalign/align) and meta.yml (mtmalign/align)
```

seems a bit weird to me.
In the clustalo module, how it was at [0d1bbc0](https://github.com/nf-core/modules/pull/4339/commits/0d1bbc05d2c5e737f5e80fc8e2f9e4b2024009fe) works (cmp https://github.com/nf-core/modules/blob/master/modules/nf-core/clustalo/align/environment.yml)

File: modules/nf-core/mtmalign/align/environment.yml [hunk 13]
```diff
@@ -0,0 +1,7 @@
+name: mtmalign/align
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/environment.yml:1]:
```suggestion
name: mtmalign_align
```

File: modules/nf-core/mtmalign/align/meta.yml [hunk 14]
```diff
@@ -0,0 +1,50 @@
+---
+# yaml-language-server: $schema=https://raw.githubusercontent.com/nf-core/modules/master/modules/yaml-schema.json
+name: "MTMALIGN_ALIGN"
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/meta.yml:3]:
```suggestion
name: "mtmalign_align"
```

**Review by luisas — CHANGES_REQUESTED** [review]

**Comment by luisas** [comment]
> Anybody know what's up with the linting there?
> 
> ```
> Conflicting process name between environment.yml (mtmalign/align) and meta.yml (mtmalign/align)
> ```
> 
> seems a bit weird to me. In the clustalo module, how it was at [0d1bbc0](https://github.com/nf-core/modules/pull/4339/commits/0d1bbc05d2c5e737f5e80fc8e2f9e4b2024009fe) works (cmp https://github.com/nf-core/modules/blob/master/modules/nf-core/clustalo/align/environment.yml)

You can see here how the tags are supposed to be - i made a couple of suggestions that should fix it https://github.com/search?q=repo%3Anf-core%2Fmodules%20clustalo&type=code

File: modules/nf-core/mtmalign/align/main.nf (hunk 4 above)
Comment (luisas) [reply modules/nf-core/mtmalign/align/main.nf]: Maybe we can then move that file into ${prefix}.pdb ? 

I think it's important to have the possibility ti have the output files named correctly

File: modules/nf-core/mtmalign/align/main.nf [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    ls *.pdb | sed s/\\ /\\n/ > input_list.txt
+    mtm-align -i input_list.txt -o ${prefix}.pdb
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/main.nf:28]: This should fix the problem with the result.pdb right?

**Review by luisas — CHANGES_REQUESTED** [review]

File: modules/nf-core/mtmalign/align/main.nf (hunk 15 above)
Comment (lrauschning) [reply modules/nf-core/mtmalign/align/main.nf:28]: yes, this fixes it in a more clean way, I think!

File: modules/nf-core/mtmalign/align/main.nf [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+    tuple val(meta), path("./mTM_result/${prefix}.aln")  , emit: alignment
+    tuple val(meta), path("./mTM_result/${prefix}.pdb")    , emit: structure
+    path "versions.yml"                                 , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/main.nf:25]:
```suggestion
    prefix = task.ext.prefix ?: "${meta.id}"
```

File: modules/nf-core/mtmalign/align/main.nf [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+    # mtm-align -v prints the wrong version 20180725, so extract it from the cosmetic output in the help message
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        mTM-align: \$( mtm-align -h | grep -e "\\(Version [[:digit:]]*\\)" | grep -oe "[[:digit:]]*" )
+    END_VERSIONS
+    """
+
+    stub:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
```
Comment (luisas) [inline modules/nf-core/mtmalign/align/main.nf:41]:
```suggestion
    prefix = task.ext.prefix ?: "${meta.id}"
```

**Review by luisas — APPROVED** [review]
This should solve the test error :)

**Pull request merged**